addons/sale/models/sale.py

```diff
@@ -248,6 +255,11 @@ def _compute_amount_undiscounted(self):
             total += line.price_subtotal + line.price_unit * ((line.discount or 0.0) / 100.0) * line.product_uom_qty  # why is there a discount in a field named amount_undiscounted ??
         self.amount_undiscounted = total
 
+    @api.depends('required_payment_percentage', 'amount_total')
+    def _compute_confirmation_amount(self):
+        for record in self:
```

Comment:
rather `order` or `sale_order`. `record` is more for mixin, when you don't know on which object you are working.